packages/components/src/card/index.tsx

```diff
@@ -30,53 +30,55 @@ type ElementProps< P, T extends TagName > = P &
 
 export type Props< T extends TagName > = ElementProps< OwnProps, T >;
 
-class Card< T extends TagName = 'div' > extends PureComponent< Props< T > > {
-	render(): JSX.Element {
-		const {
-			children,
-			className,
-			compact,
-			displayAsLink,
-			highlight,
-			tagName = 'div',
-			href,
-			target,
-			...props
-		} = this.props;
+const Card = < T extends TagName = 'div' >(
+	{
+		children,
+		className,
+		compact,
+		displayAsLink,
+		highlight,
+		tagName = 'div',
+		href,
+		target,
+		...props
+	}: Props< T >,
+	forwardedRef: Ref< any > // eslint-disable-line @typescript-eslint/no-explicit-any
+) => {
+	const elementClass = classNames(
+		'card',
+		className,
+		{
+			'is-card-link': displayAsLink || href,
+			'is-clickable': props.onClick,
+			'is-compact': compact,
+			'is-highlight': highlight,
+		},
+		highlight ? 'is-' + highlight : false
+	);
 
-		const elementClass = classNames(
-			'card',
-			className,
-			{
-				'is-card-link': displayAsLink || href,
-				'is-clickable': this.props.onClick,
-				'is-compact': compact,
-				'is-highlight': highlight,
-			},
-			highlight ? 'is-' + highlight : false
-		);
-
-		return href ? (
-			<a { ...props } href={ href } target={ target } className={ elementClass }>
-				<Gridicon className="card__link-indicator" icon={ target ? 'external' : 'chevron-right' } />
+	return href ? (
+		<a { ...props } href={ href } target={ target } className={ elementClass } ref={ forwardedRef }>
+			<Gridicon className="card__link-indicator" icon={ target ? 'external' : 'chevron-right' } />
+			{ children }
+		</a>
+	) : (
+		React.createElement(
+			tagName,
+			{ ...props, className: elementClass, ref: forwardedRef },
+			<>
+				{ displayAsLink && (
+					<Gridicon
+						className="card__link-indicator"
+						icon={ target ? 'external' : 'chevron-right' }
+					/>
+				) }
 				{ children }
-			</a>
-		) : (
-			React.createElement(
-				tagName,
-				{ ...props, className: elementClass },
-				<>
-					{ displayAsLink && (
-						<Gridicon
-							className="card__link-indicator"
-							icon={ target ? 'external' : 'chevron-right' }
-						/>
-					) }
-					{ children }
-				</>
-			)
-		);
-	}
-}
+			</>
+		)
+	);
+};
+
+const ForwardedRefCard = React.forwardRef( Card );
+ForwardedRefCard.displayName = 'ForwardRef(Card)';
```

Comment:
Do we need the `ForwardRef()` part in the component name? I'm seeing the exact opposite [being done in Gutenberg](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/panel/body.js#L131) where we alter the name to be the original one.

Reply:
We don't, I can change it to just be the name of the component, but when inspecting the component tree it will leave out this potentially helpful information 🤷🏼‍♀️

Reply:
I don't have strong feelings towards making it just `Card`; just felt it isn't a common pattern I'm used to seeing. My preference would be to make it just `Card`, as I believe we can't (and shouldn't) express all the useful information about a component's specificities (caching, hooks, HoCs, etc) in the name, but I won't object if you prefer to keep it that way 😉

Reply:
I did a little bit of testing and found that this toy card component:
```jsx
const Card = React.forwardRef((props, ref) => (
  <div ref={ref}>{props.children}</div>
));
Card.displayName = "Card";
```
is shown in devtools as:

<img width="339" alt="Screenshot 2021-03-11 at 7 15 02" src="https://user-images.githubusercontent.com/664258/110745155-24fa5900-823b-11eb-8f1b-65a88c2bfa2d.png">

I.e., the `ForwardRef` info is already there. There doesn't need to be another `ForwardRef( Card )`.

Without the `displayName` defined, it would be shown as `Anonymous` instead of `Card`.

The `Card( props, ref )` function defined here is not a real component -- it's a `forwardRef` callback. The return value of `forwardRef` is the only thing in this file that is a component and deserves a `displayName`.